index.bs

```diff
@@ -137,7 +165,34 @@ Coordinates</a></code> interface of the Geolocation API are the following:
 </div>
 
 To <dfn>construct a <a>Geolocation Sensor</a> object</dfn> the user agent must invoke the
-[=construct a Sensor object=] abstract operation.
+[=construct a Sensor object=] abstract operation for the {{GeolocationSensor}} interface.
+
+### GeolocationSensor.read() ### {#geolocationsensor-read}
+
+The {{GeolocationSensor/read()}} method, when called, must run the following algorithm:
+
+<div algorithm="read">
+<!-- https://dom.spec.whatwg.org/#abortcontroller-api-integration -->
+
+    : input
+    :: |options|, a {{ReadOptions}} object.
+    : output
+    :: |p|, a promise.
+
+    1. Let |p| be a new promise.
+    1. Let |options| be the first argument.
+    1. Let |signal| be the |options|' dictionary member of the same name if present, or null otherwise.
+    1. Run these steps <a>in parallel</a>:
+      1. Let |geo| be the result of invoking <a>construct a Geolocation Sensor object</a> with |options|.
+      1. Invoke |geo|.{{Sensor/start()}}.
+      1. If <a>notify error</a> is invoked with |geo| as input, reject |p| with the {{DOMException}} passed as input to <a>notify error</a>.
+      1. If |signal|'s <a>aborted flag</a> is set, then reject |p| with an "{{AbortError!!exception}}" {{DOMException}}.
```

Comment:
There is no return (abort of algorithm), therefore, promise is rejected twice. Here, and in the following `if` branch.

It would be better to move signal setup code before start (before run in parallel), so that we reject early and abort steps if signal's abort flag is set.